plugins/linux-pipewire/pipewire.c

```diff
@@ -396,42 +462,87 @@ static void init_format_info(obs_pipewire_data *obs_pw)
 	bfree(drm_formats);
 }
 
-static void clear_format_info(obs_pipewire_data *obs_pw)
+static void init_format_info_output(obs_pipewire_stream *obs_pw_stream)
+{
+	da_init(obs_pw_stream->format_info);
+
+	for (size_t i = 0; i < N_SUPPORTED_FORMATS_OUTPUT; i++) {
+		struct format_info *info, tmp;
+		if (!lookup_format_info_from_spa_format(
+			    supported_formats_output[i], &tmp))
+			continue;
+
+		info = da_push_back_new(obs_pw_stream->format_info);
+		da_init(info->modifiers);
+		info->spa_format = tmp.spa_format;
+		info->drm_format = tmp.drm_format;
+	}
+}
+
+static void init_format_info(obs_pipewire_stream *obs_pw_stream)
+{
+	if (obs_pw_stream->obs_data.type == OBS_PIPEWIRE_STREAM_TYPE_OUTPUT) {
+		init_format_info_output(obs_pw_stream);
+		return;
+	}
+
+	uint32_t output_flags;
+
+	output_flags = obs_source_get_output_flags(obs_pw_stream->source);
```

Comment:
I had to do this to make it rework.
```suggestion
	output_flags = obs_source_get_output_flags(obs_pw_stream->obs_data.source);
```